### ISSUE TITLE
Is this supposed to work on Windows/WSL2?

### ISSUE THREAD

**Issue by escherstair** [commontorizon/torizonEmulatorManager#1]
I read that Torizon Emulator Manager runs on Linux, and it depends on QEMU and Docker (at least).
The tool itself seems cool and useful.
I wonder if it is supposed to work on Windows (native or under WSL2)

**Comment by microhobby** [comment]
hey @escherstair , yeah it should work. Follow the instructions from https://github.com/commontorizon/torizonEmulatorManager?tab=readme-ov-file#installation

Let me know if this works for you.

**Comment by escherstair** [comment]
Hi @microhobby 
I've just installed, but when I start it I get the following error

```bash
user@HTGRF:~$ torizon-emulator-manager
Hello Torizon!
Starting Emulator
WARN[0000] /home/torizon/app/assets/docker-compose.yml: `version` is obsolete
[+] Pulling 7/1
 ✔ emulator Pulled          148.5s 7MB Pulling 148.5s
WARN[0000] /home/torizon/app/assets/docker-compose.yml: `version` is obsolete
[+] Creating 1/1
 ✔ Container assets-dhcp-1  Created0.4s
[+] Running 1/1
 ✔ Container assets-dhcp-1  Started0.2s
Could not access KVM kernel module: No such file or directory
qemu-system-x86_64: failed to initialize kvm: No such file or directory
WARN[0000] /home/torizon/app/assets/docker-compose.yml: `version` is obsolete
[+] Running 1/1
 ✔ Container assets-dhcp-1  Removed11.1s
```

Should I install `qemu-kvm` or some other pack in Ubuntu under WSL2?

**Comment by microhobby** [comment]
not really, all the dependencies should be on the container. Maybe the kernel version from your WSL2 is old one.
Could you report the result from the follow command from Windows side (cmd, or powershell sessions):

```
wsl --version
```

**Comment by escherstair** [comment]
Hi @microhobby 
Here is the log
```bash
wsl --version
Versione WSL: 2.0.14.0
Versione kernel: 5.15.133.1-1
Versione WSLg: 1.0.59
Versione MSRDC: 1.2.4677
Versione Direct3D: 1.611.1-81528511
Versione DXCore: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Versione di Windows: 10.0.19045.4291
```
Then I updated to
```bash
wsl --version
Versione WSL: 2.1.5.0
Versione kernel: 5.15.146.1-2
Versione WSLg: 1.0.60
Versione MSRDC: 1.2.5105
Versione Direct3D: 1.611.1-81528511
Versione DXCore: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Versione di Windows: 10.0.19045.4291
```
but I got the same error.

**Comment by microhobby** [comment]
@escherstair weird, the latest version should have the KVM enabled. Could you please check if there is a `/dev/kvm` in your WSL 2 system?

Also, check if you have a `.wslconfig` file on the root of your `C:\users\<yourUserName>\` with the following:

```
[wsl2]
nestedVirtualization=false
```
 The default is true, so, if there is nothing about it on your user folder it should be ok 🤔.

**Comment by escherstair** [comment]
@microhobby 
I dont' have `/dev/kvm` in my WSL2.
And I don't have any `.wslconfig` file on the root of my `C:\users\<yourUserName>\`
But I read [here](https://learn.microsoft.com/en-us/windows/wsl/wsl-config#wslconfig)
![immagine](https://github.com/commontorizon/torizonEmulatorManager/assets/22594460/c941e92b-78d6-4554-85e5-7e60c70716de)
and I see it's Windows 11 only.
And I have Windows 10.

I found [this](https://github.com/sickcodes/Docker-OSX#id-like-to-run-docker-osx-on-windows) and so I run this command
```bash
sudo apt -y install bridge-utils cpu-checker libvirt-clients libvirt-daemon qemu qemu-kvm
```
Now `/dev/kvm` exists in WSL2, but running `kvm-ok` gives
```
INFO: Your CPU does not support KVM extensions
KVM acceleration can NOT be used
```

Nothing changes if I create a `.wslconfig` with
```
[wsl2]
nestedVirtualization=false
```

Does torizonEmulatorManager require Windows 11?

**Comment by microhobby** [comment]
Hey @escherstair yeah, unfortunately nested virtualization with WSL 2 only works on Windows 11 😢.

Would upgrading to Win 11 be something you should consider? If not, could you give feedback on the causes?

There is still a way to run the emulator without `/dev/kvm`, but performance is greatly affected. I updated the application to automatically check if `/dev/kvm` is available and then fallback to x86-64 emulator if not available. Let me know if this works.

**Comment by escherstair** [comment]
Hi @microhobby 
I won't update to Win11 until [this issue](https://github.com/microsoft/WSL/issues/10268) is solved (if ever).
I opened the topic on Toradex community too (different nickname here, but on Github it was not available 😢)
Or my IT won't allow me to run Win10 anymore.

**Comment by microhobby** [comment]
got it, ok, thanks to share @escherstair

**Comment by escherstair** [comment]
I've just tried on another Windows 11 machine and it works but it must be run with `sudo` rights
`sudo torizon-emulator-manager`

I see that docker images are downloaded and startedas containers (QEMU + a predefined version of Torizon OS).
Is there some documentaion on how to select a different flavor of Torizon OS, etc...?

**Comment by escherstair** [comment]
Hi @microhobby,
sorry for the late answer

> There is still a way to run the emulator without `/dev/kvm`, but performance is greatly affected. I updated the application to automatically check if `/dev/kvm` is available and then fallback to x86-64 emulator if not available. Let me know if this works.

I've just tried but it doesn't work
Now I see `/dev/kvm` in my WSL2 on Windows 10 (don't know why), but
here is the log

``` bash
torizon-emulator-manager
access control disabled, clients can connect from any host
Initializing ...
Starting Emulator
WARN[0000] The "NO_KVM" variable is not set. Defaulting to a blank string.
WARN[0000] /home/torizon/app/assets/docker-compose.yml: `version` is obsolete
[+] Pulling 1/1
 ✔ emulator Pulled 1.1s
WARN[0000] The "NO_KVM" variable is not set. Defaulting to a blank string.
WARN[0000] /home/torizon/app/assets/docker-compose.yml: `version` is obsolete
[+] Creating 1/0
 ✔ Container assets-dhcp-1  Created0.1s
[+] Running 1/0
 ✔ Container assets-dhcp-1  Started0.1s
Could not access KVM kernel module: No such device
qemu-system-x86_64: failed to initialize kvm: No such device
WARN[0000] The "NO_KVM" variable is not set. Defaulting to a blank string.
WARN[0000] /home/torizon/app/assets/docker-compose.yml: `version` is obsolete
[+] Running 1/1
 ✔ Container assets-dhcp-1  Removed10.2s
```
